### PR TITLE
Feature/add sh19 payment report

### PR DESCRIPTION
**Pull request by PhilipGlover**
Add a new payment report for SH19 transaction to be issued to a separate email address

BI-6086

Associated config PR's:
- https://github.com/companieshouse/chs-configs/pull/1552
- https://github.com/companieshouse/docker-chs-development/pull/226

Screenshot 2021-06-11 at 12 30 40

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/efs/api/email/config/PaymentReportEmailConfig.java [hunk 1]
```diff
@@ -29,6 +30,14 @@ public void setScottishEmailAddress(String scottishEmailAddress) {
         this.scottishEmailAddress = scottishEmailAddress;
     }
 
+    public String getSpecialCapitalEmailAddress() {
+        return specialCapitalEmailAddress;
+    }
+
+    public void setSpecialCapitalEmailAddress(final String specialCapitalEmailAddress) {
```
Comment (hepsimo) [inline src/main/java/uk/gov/companieshouse/efs/api/email/config/PaymentReportEmailConfig.java:37]: Please add unit test coverage

File: src/main/java/uk/gov/companieshouse/efs/api/email/mapper/PaymentReportEmailMapper.java [hunk 2]
```diff
@@ -39,11 +39,22 @@ public PaymentReportEmailMapper(final PaymentReportEmailConfig config, Identifie
...
             .withFileName(model.getFileName())
             .witHasNoPaymentTransactions(model.getHasNoPaymentTransactions())
             .build();
     }
+
+    private String getEmailAddressForReportFilename(final PaymentReportEmailModel model) {
+        String filename = model.getFileName();
+        if (filename.contains("Scottish")) {
+            return config.getScottishEmailAddress();
+        } else if (filename.contains("SH19")) {
```
Comment (hepsimo) [inline src/main/java/uk/gov/companieshouse/efs/api/email/mapper/PaymentReportEmailMapper.java:54]: Please add unit test coverage for "SH19" case

**Review by hepsimo — CHANGES_REQUESTED** [review]
Looks good except for a couple of minor missed test coverage spots

File: src/main/java/uk/gov/companieshouse/efs/api/email/config/PaymentReportEmailConfig.java (hunk 1 above)
Comment (PhilipGlover) [reply src/main/java/uk/gov/companieshouse/efs/api/email/config/PaymentReportEmailConfig.java:37]: this one was easy - added `setGetSpecialCapitalEmailAddress()`

File: src/main/java/uk/gov/companieshouse/efs/api/email/mapper/PaymentReportEmailMapper.java (hunk 2 above)
Comment (PhilipGlover) [reply src/main/java/uk/gov/companieshouse/efs/api/email/mapper/PaymentReportEmailMapper.java:54]: this took longer and the supporting methods now have a number of conditions to handle the expected data - would now be easier to add any further tests for new reports

**Review by hepsimo — APPROVED** [review]
Changes look OK

**Pull request merged**